picard/script/__init__.py

```diff
@@ -101,3 +114,236 @@ def enabled_tagger_scripts_texts():
+
+        Args:
+            **kwargs: Properties to update in the form `property=value`.
+        """
+        self._update_from_dict(kwargs)
+
+    def _update_from_dict(self, settings):
+        """Updates the values of the properties based on the contents of the dictionary provided.
+        Properties in the `settings` dictionary that are not found in the script object are ignored.
+
+        Args:
+            settings (dict): The settings to update.
+        """
+        updated = False
+        for key, value in settings.items():
+            if value is not None and hasattr(self, key):
+                if isinstance(value, str):
+                    value = value.strip()
+                setattr(self, key, value)
+                if key not in self._last_updated_ignore_list:
+                    updated = True
+        # Don't overwrite `last_updated` with a generated value if a last_updated value has been provided.
+        if updated and 'last_updated' not in settings:
+            self.update_last_updated()
+
+    def copy(self):
+        """Create a copy of the current script object with updated title and last updated attributes.
+        """
+        new_object = deepcopy(self)
+        new_object.update_script_setting(title=self.title + N_(" (Copy)"))
+        new_object._set_new_id()
+        return new_object
+
+    def to_json(self, indented=False):
+        """Converts the properties of the script object to a JSON formatted string.  Note that only property
+        names listed in `JSON_PROPERTIES` will be included in the output.
+
+        Args:
+            indented (bool): Indent the output.
+
+        Returns:
+            str: The properties of the script object formatted as a JSON string.
+        """
+        items = {key: getattr(self, key) for key in dir(self) if key in self.JSON_OUTPUT}
+        if indented:
+            return json.dumps(items, indent=4, sort_keys=True)
+        return json.dumps(items, sort_keys=True)
+
+    @classmethod
+    def create_from_json(cls, json_string):
+        """Creates an instance based on the contents of the JSON string provided.
+        Properties in the JSON string that are not found in the script object are ignored.
+
+        Args:
+            json_string (str): JSON string containing the property settings.
+
+        Returns:
+            object: An instance of the class, populated from the property settings in the JSON string.
+        """
+        new_object = cls()
+        new_object.update_from_json(json_string)
+        return new_object
+
+    def update_from_json(self, json_string):
+        """Updates the values of the properties based on the contents of the JSON string provided.
+        Properties in the JSON string that are not found in the script object are ignored.
+
+        Args:
+            json_string (str): JSON string containing the property settings.
+        """
+        settings = json.loads(json_string)
+        self._update_from_dict(settings)
```

Comment:
Since that's pretty simple & short, I think we can drop the intermediate variable here:

```python
self._update_from_dict(json.loads(json_string))
```